### ISSUE TITLE
Streamline pypi deployment process

### ISSUE THREAD

**Issue by pdiazvargas** [eabglobal/juniper#12]
## What
Streamline the process in which juniper is build and push to pypi. Refactor and take a second look at the `make-release.py` which was taken straight out of flask. This was a great starting point, but there are multiple points of improvements.

In the new version, we need to make sure to correctly update the value of `juniper.__init__:__version__`. Also, it would be nice to be able to deploy the code to the test pypi environment first.

## How
Ideally this should be an automated step. As part of the deployment process, whenever a new feature is merged into master, the upload of the library to pypi should occur.
For now, it would be ok for it to be a manual step that the maintainer takes right after the PR is merged to master. Before merging anything to master, the maintainer should be able to deploy the new code to the pypi testing environment as a way to make sure that everything will be correctly uploaded.

Once the feature is merged, the script can be executed to update the version of the file as well as update the project in pypi.

## Why
Streamline and enhance manual processes will prevent errors in the long run.
